lib/request_started_on_middleware.rb

```diff
@@ -0,0 +1,57 @@
+# Add to config/application.rb:
+#
+#     config.middleware.use 'RequestStartedOnMiddleware'
+#
+class RequestStartedOnMiddleware
+  def initialize(app)
+    @app = app
+  end
+
+  def call(env)
+    start_request(env['PATH_INFO'], Time.now.utc)
+    @app.call(env)
+  ensure
+    complete_request
+  end
+
+  def start_request(path, started_on)
+    Thread.current[:current_request] = path
+    Thread.current[:current_request_started_on] = started_on
+  end
+
+  def complete_request
+    Thread.current[:current_request] = nil
+    Thread.current[:current_request_started_on] = nil
+  end
+
+  def self.long_running_requests
+    requests = []
+    timed_out_request_started_on = request_timeout.ago.utc
+
+    relevant_thread_list.each do |thread|
+      request    = thread[:current_request]
+      started_on = thread[:current_request_started_on]
+
+      # There's a race condition where the complete_request method runs in another
+      # thread after we set one or more of the above local variables. The fallout
+      # of this is we return a false positive for a request that finished very close
+      # to the 2 minute timeout.
```

Comment:
Just a note for the future:  This becomes a bigger issue if we were "react to a hung request" as you suggested [here](https://github.com/ManageIQ/manageiq/pull/17842#issuecomment-412618419), but for now, this really isn't a big deal.

Logging something that was a "close to a 2 min request" is going to anger just about no one (accept @kbrock, who probably thinks we log too much as it is... which is fair, but not relevant to my point).

Reply:
@NickLaMuro 😆 🤣 @kbrock